src/Microsoft.Data.SqlClient/tests/Common/LocalAppContextSwitchesHelper.cs

```diff
@@ -0,0 +1,355 @@
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+using Xunit;
+
+namespace Microsoft.Data.SqlClient.Tests.Common;
+
+// This class provides read/write access to LocalAppContextSwitches values
+// for the duration of a test.  It is intended to be constructed at the start
+// of a test and disposed at the end.  It captures the original values of
+// the switches and restores them when disposed.
+//
+// This follows the RAII pattern to ensure that the switches are always
+// restored, which is important for global state like LocalAppContextSwitches.
+//
+// https://en.wikipedia.org/wiki/Resource_acquisition_is_initialization
+//
+public sealed class LocalAppContextSwitchesHelper : IDisposable
+{
+    #region Public Types
+
+    // This enum is used to represent the state of a switch.
+    //
+    // It is a copy of the Tristate enum from LocalAppContextSwitches.
+    //
+    public enum Tristate : byte
+    {
+        NotInitialized = 0,
+        False = 1,
+        True = 2
+    }
+
+    #endregion
+
+    #region Construction
+
+    // Construct to capture all existing switch values.
+    //
+    // Fails the test if any values cannot be captured.
+    //
+    public LocalAppContextSwitchesHelper()
+    {
+        // Acquire a handle to the LocalAppContextSwitches type.
+        var assembly = typeof(SqlCommandBuilder).Assembly;
+        var switchesType = assembly.GetType(
+            "Microsoft.Data.SqlClient.LocalAppContextSwitches");
+        if (switchesType == null)
+        {
+            Assert.Fail("Unable to find LocalAppContextSwitches type.");
```

Comment:
My philosophy on unit tests is that `Assert` should only be called for what is being tested. So, in this case, we should just throw an exception. It will fail the test anyhow, but imo it makes it clearer that it's something to do with the infra rather than the test conditions failing.